src/pages/resultsView/ResultsViewPageHelpers.ts

```diff
@@ -85,12 +85,14 @@ export function substitutePhysicalStudiesForVirtualStudies(cancerStudyIds:string
 export function populateSampleSpecificationsFromVirtualStudies(samplesSpecification:SamplesSpecificationElement[], virtualStudies:VirtualStudy[]){
 
     const virtualStudiesKeyedById = _.keyBy(virtualStudies,(virtualStudy)=>virtualStudy.id);
-
+    const samplesSpecificationKeyedByStudyId = _.keyBy(samplesSpecification,(spec)=>spec.studyId);
     // remove specs for virtual studies (since they mean nothing to api)
     // and then populate with ids
     samplesSpecification = _.filter(samplesSpecification,(spec)=>!virtualStudiesKeyedById[spec.studyId]);
```

Comment:
probably should be sampleSpecifications.  Plural should be at end because it's an array of specifications.